lib/hosts.nix

```
@@ -13,7 +13,17 @@
           self.nixosModules.users
         ]
         ++ base
-        ++ extraModules;
+        ++ extraModules
+        ++ [
+          ({...}: {
+            networking.hostName = hostname;
+            networking.domain = domain;
+
+            imports = [
+              "${self}/hosts/${hostname}.${domain}"
```

can't we use
```suggestion
              "${self}/hosts/${config.networking.fqdn}"
```
here?

---

No, that's an infinite recursion.